modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -788,6 +790,21 @@ private ObjectDefinition _addObjectDefinition(
 				LanguageUtil.get(LocaleUtil.getDefault(), "create-date")),
 			"createDate", false, false);
 
+		if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-158821")) &&
+			!objectDefinition.isSystem()) {
+
+			_objectFieldLocalService.addSystemObjectField(
+				userId, objectDefinition.getObjectDefinitionId(),
+				ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+				ObjectEntryTable.INSTANCE.externalReferenceCode.getName(),
+				dbTableName, ObjectFieldConstants.DB_TYPE_STRING, null, false,
```

Comment:
Hi @seltong,

For this case you can just replace `dbTableName` to `ObjectEntryTable.INSTANCE.getTableName()`